jetty-server/src/main/java/org/eclipse/jetty/server/handler/ThreadLimitHandler.java

```diff
@@ -0,0 +1,459 @@
+ * headers are likely from untrusted client side proxies.</li>
+ * <li>If multiple instances of a forwarded header are provided, this
+ * handler will use the right-most instance, which will have been set from
+ * the trusted local proxy</li>
+ * </ul>
+ * Requests in excess of the limit will be asynchronously suspended until
+ * a thread is available.  
+ * <p>This is a simpler alternative to DosFilter</p>
+ */
+public class ThreadLimitHandler extends HandlerWrapper
+{
+    private static final Logger LOG = Log.getLogger(ThreadLimitHandler.class);
+
+    private final static String REMOTE = "o.e.j.s.h.TLH.REMOTE";
+    private final static String PASS = "o.e.j.s.h.TLH.PASS";
+    private final boolean _rfc7239;
+    private final String _forwardedHeader;
+    private final IncludeExcludeSet<String, InetAddress> _includeExcludeSet = new IncludeExcludeSet<>(InetAddressSet.class);
+    private final ConcurrentMap<String, Remote> _remotes = new ConcurrentHashMap<>();
+    private volatile boolean _enabled;
+    private long _blockForMs=0;
+    private int _threadLimit=10;
+
+    public ThreadLimitHandler()
+    {
+        this(null,false);
+    }
+    
+    public ThreadLimitHandler(@Name("forwardedHeader") String forwardedHeader)
+    {
+        this(forwardedHeader,HttpHeader.FORWARDED.is(forwardedHeader));
+    }
+
+    public ThreadLimitHandler(@Name("forwardedHeader") String forwardedHeader, @Name("rfc7239")  boolean rfc7239)
+    {
+        super();
+        _rfc7239 = rfc7239;
+        _forwardedHeader = forwardedHeader;
+        _enabled = true;
+    }
+    
+    @Override
+    protected void doStart() throws Exception
+    {
+        if (_threadLimit<0)
+            _enabled=false;
+        else if (_threadLimit==0)
+        {
+            ThreadPool threadPool = getServer().getThreadPool();
+            if (threadPool instanceof SizedThreadPool)
+                _threadLimit=Math.max(10,((SizedThreadPool)threadPool).getMaxThreads()/10);
+            else
+                _threadLimit=10;
+        }
+        super.doStart();
+    }
+
+    public boolean isEnabled()
+    {
+        return _enabled;
+    }
+
+    @ManagedAttribute("enable/disable thread limits")
+    public void setEnabled(boolean enabled)
+    {
+        _enabled = enabled;
+    }
+
+    public long getBlockForMs()
+    {
+        return _blockForMs;
+    }
+
+    @ManagedAttribute("Period in ms to wait synchronously for available thread")
+    public void setBlockForMs(long blockForMs)
+    {
+        _blockForMs = blockForMs;
+    }
+
+    public int getThreadLimit()
+    {
+        return _threadLimit;
+    }
+
+    @ManagedAttribute("The maximum threads that can be dispatched per remote IP")
+    public void setThreadLimit(int threadLimit)
+    {
+        _threadLimit = threadLimit;
+        if (_threadLimit<0)
+            _enabled=false;
+    }
+
+    @ManagedOperation("Include IP in thread limits")
+    public void include(String... inetAddressPattern)
```

Comment:
I don't think JMX is able to support varargs methods. Have you tried in JMC if it is possible to actually call these methods via JMX ? Otherwise I think we need to add `include(String s)`.